src/components/uploadSiteImageButton/index.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { Modal } from 'antd';
+import { useTranslation } from 'react-i18next';
+import { site } from '../../constants';
+import { n } from '../../utils/stringFormat';
+import { GreenButton, StyledClose, SubmitButton } from '../themedComponents';
+import { InboxOutlined } from '@ant-design/icons';
+import { message, Upload } from 'antd';
+import { UploadProps } from 'antd/lib/upload/interface';
+import { LIGHT_GREEN, LIGHT_GREY } from '../../utils/colors';
+import protectedApiClient from '../../api/protectedApiClient';
+
+const { Dragger } = Upload;
+
+const StyledInboxOutline = styled(InboxOutlined)`
+  color: black;
+`;
+
+const ConfirmUpload = styled(SubmitButton)`
+  margin: 10px;
+  padding-left: 10px;
+
+  & :hover {
+    background-color: ${LIGHT_GREY};
+  }
+`;
+
+interface UploadImageProps {
+  readonly siteEntryId: number;
+}
+
+const UploadSiteImageButton: React.FC<UploadImageProps> = ({ siteEntryId }) => {
+  const { t } = useTranslation(n(site, ['treeInfo']), {
+    nsMode: 'fallback',
+  });
+  const [showMenu, setShowMenu] = useState(false);
+  let imageToUpload: string | ArrayBuffer | null;
+
+  const props: UploadProps = {
+    name: 'file',
+    multiple: false,
```

Comment:
💭 Do we want to allow users to upload multiple images at a time? It'd be nice quality of life feature for users, but it'd take a bit more work to upload each image and handle individual errors. If not, I think we should add the `maxCount` prop as 1 (more info in the [component doc](https://ant.design/components/upload#api)) so it's clear to users that they can only upload one image at time.